### ISSUE TITLE
google API account carddev template

### ISSUE THREAD

**Issue by KelvinY-hen** [natelindev/tsdav#175]
is there any documentation or sample on how to use carddav on google api? ive been facing invalid credentials problem for a while now and following steps all according to my information (which only related to caldav)

const client =  new DAVClient({
    serverUrl: 'https://www.googleapis.com/carddav/v1/principals/YOUREMAIL',
    credentials: {
      tokenUrl: 'https://oauth2.googleapis.com/token',
      clientId: 'OAuth 2.0 Client IDs',
      clientSecret: 'OAuth 2.0 Client SECRET',
      redirectUrl: "http://localhost:3001",
      authorizationCode:"OAuth 2.0 AUTHORIZATION"
    },
    authMethod: 'Oauth',
    defaultAccountType: 'cardddav',
  });

**Comment by natelindev** [comment]
let me test it first and then get back to you, carddav functions of this lib are not as well tested as caldav functions.

**Comment by KelvinY-hen** [comment]
@natelindev any update regarding the carddav for google?
\

**Comment by natelindev** [comment]
Sorry I was too busy last week. I'll look into it later today.

**Comment by natelindev** [comment]
Sorry the doc is a bit misleading.

for google auth you should probably not use `authorizationCode` as credentials.
the correct way to use google auth is like:
```
{
    tokenUrl: 'https://accounts.google.com/o/oauth2/token',
    username: 'YOUR_EMAIL_ADDRESS',
    refreshToken: 'YOUR_REFRESH_TOKEN_WITH_CALDAV_PERMISSION',
    clientId: 'YOUR_CLIENT_ID',
    clientSecret: 'YOUR_CLIENT_SECRET',
}
```
you could obtian refresh token using `fetchOauthTokens`
or any other way you see fit.

I'll update the docs.

**Comment by natelindev** [comment]
fixed in v2.0.9
